ambry-store/src/main/java/com/github/ambry/store/StorageManager.java

```diff
@@ -756,31 +759,32 @@ public void onPartitionBecomeDroppedFromOffline(String partitionName) {
       // the store could be started if it failed on decommission last time. Helix may directly reset it to OFFLINE
       // without stopping it)
       BlobStore store = (BlobStore) getStore(replica.getPartitionId(), true);
-
-      // 1. Check if the store is recovering from decommission or directly transitioning from OFFLINE to DROPPED in
-      // full-auto (i.e. if this replica has been reassigned to a different host when it is down, Helix may directly
-      // transition the replica from OFFLINE -> DROPPED without going through OFFLINE -> BOOTSTRAP -> STANDBY ->
-      // INACTIVE -> OFFLINE -> DROPPED steps). If so, go through decommission steps to make sure peer replicas are
-      // caught up with local replica and we update DataNodeConfig in Helix.
-      if (store.recoverFromDecommission() || (clusterMap.isDataNodeInFullAutoMode(replica.getDataNodeId())
-          && store.getPreviousState() == ReplicaState.OFFLINE && !isReplicaOnFailedDisk(replica))) {
-        try {
-          resumeDecommission(partitionName);
-        } catch (Exception e) {
-          logger.error("Exception occurs when resuming decommission on replica {} with error msg: {}", replica,
-              e.getMessage());
-          metrics.resumeDecommissionErrorCount.inc();
-          throw new StateTransitionException(
-              "Exception occurred when resuming decommission on replica " + partitionName, ReplicaOperationFailure);
+      if (store == null && isReplicaOnFailedDisk(replica)) {
+        logger.info("Replica is in a failed disk, blob store not started. skip");
+      } else {
+        // 1. Check if the store is recovering from decommission or directly transitioning from OFFLINE to DROPPED in
+        // full-auto (i.e. if this replica has been reassigned to a different host when it is down, Helix may directly
+        // transition the replica from OFFLINE -> DROPPED without going through OFFLINE -> BOOTSTRAP -> STANDBY ->
+        // INACTIVE -> OFFLINE -> DROPPED steps). If so, go through decommission steps to make sure peer replicas are
+        // caught up with local replica and we update DataNodeConfig in Helix.
+        if (store.recoverFromDecommission() || (clusterMap.isDataNodeInFullAutoMode(replica.getDataNodeId())
```

Comment:
Minor: Would we need null check here? It seems like we could come here if store == null and isReplicaOnFailedDisk(replica) = false.

Reply:
I don't think we need to. When deal with disk failures, the replicas has already removed from partitionNameToReplicaId, so the replicaId will be null and the execution would enter the if statement above and return in the if statement.